pkg/gardenlet/controller/controllerinstallation/controllerinstallation.go

```diff
@@ -64,7 +65,7 @@ type Controller struct {
 }
 
 // NewController instantiates a new ControllerInstallation controller.
-func NewController(k8sGardenClient kubernetes.Interface, gardenCoreInformerFactory gardencoreinformers.SharedInformerFactory, config *config.ControllerManagerConfiguration, recorder record.EventRecorder, gardenNamespace *corev1.Namespace) *Controller {
+func NewController(k8sGardenClient kubernetes.Interface, gardenCoreInformerFactory gardencoreinformers.SharedInformerFactory, config *config.GardenletConfiguration, recorder record.EventRecorder, gardenNamespace *corev1.Namespace) *Controller {
```

Comment:
I don’t get the benefit of doing this complex rework. The gardenlet needs to talk to the garden cluster anyways, so it can fetch the installations from there. We can discuss separately in the future, by I would not consider this important for now. The PR is already large enough.

Reply:
Fine